service/src/main/scala/com/yahoo/maha/service/curators/DrilldownCurator.scala

```diff
@@ -156,14 +157,78 @@ class DrilldownCurator (override val requestModelValidator: CuratorRequestModelV
                                             mahaService: MahaService,
                                             mahaRequestLogBuilder: CuratorMahaRequestLogBuilder,
                                             primaryKeyAlias: String,
-                                            drilldownConfig: DrilldownConfig): ReportingRequest = {
-    val (rm, fields) : (RequestModel, IndexedSeq[Field]) = validateReportingRequest(registryName, bucketParams, reportingRequest, mahaService, mahaRequestLogBuilder)
+                                            drilldownConfig: DrilldownConfig,
+                                            mahaRequestContext: MahaRequestContext): ReportingRequest = {
+    val (_, fields) : (RequestModel, IndexedSeq[Field]) = validateReportingRequest(registryName, bucketParams, reportingRequest, mahaService, mahaRequestLogBuilder)
     val primaryField : Field = Field(primaryKeyAlias, None, None)
 
-    val rr = drilldownReportingRequest(reportingRequest, fields, primaryField, drilldownConfig)
+    val fields_reduced : IndexedSeq[Field] = removeInvalidFactAliases(registryName
+    , mahaService.getMahaServiceConfig
+    , drilldownConfig
+    , fields
+    , mahaRequestContext)
+
+    val rr = drilldownReportingRequest(reportingRequest, fields_reduced, primaryField, drilldownConfig)
     rr
   }
 
+  def removeInvalidFactAliases(registryName: String,
+                               mahaServiceConfig: MahaServiceConfig,
+                               drilldownConfig: DrilldownConfig,
+                               factFields: IndexedSeq[Field],
+                               context: MahaRequestContext): IndexedSeq[Field] = {
+    if(drilldownConfig.cube.isEmpty){
+      factFields
+    }
+    else{
+      val registryConfig: RegistryConfig = mahaServiceConfig.registry(registryName)
+      val factMap: Map[(String, Int), PublicFact] = registryConfig.registry.factMap
+      val selector: BucketSelector = registryConfig.bucketSelector
+
+      val selectedRevisionTry = Try(registryConfig.registry.defaultPublicFactRevisionMap(drilldownConfig.cube))
+
+      val selectedRevision: Option[Int] = if(selectedRevisionTry.isFailure){
+        val bucketSelectedTry : Try[BucketSelected] = selector.selectBuckets(
+          drilldownConfig.cube
+          , context.bucketParams.copy(forceRevision = None)
+        )
+
+        bucketSelectedTry match {
+          case Success(bucketSelected) =>
+            Some(bucketSelected.revision)
+
+          case Failure(t) =>
+            logger.info("Failed to select valid bucket for fact " + t)
+            throw new IllegalArgumentException(t.getMessage)
+        }
+
+
+      }
+      else{
+          selectedRevisionTry.toOption
+      }
+
+      var factFieldsReduced: IndexedSeq[Field] = IndexedSeq.empty
```

Comment:
Why do you need a var here?  Why not just do single assignment with val since you only set it once?
```
val factFieldsReduced: IndexedSeq[Field] = factFields.filter (field =>
  pubFact.columnsByAlias.contains (field.field)
)
```